File: spring-boot-starter-breuninger-togglz/src/main/java/com/breuninger/boot/togglz/configuration/TogglzProperties.java
package com.breuninger.boot.togglz.configuration;

import javax.validation.Valid;
import javax.validation.constraints.Min;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@ConfigurationProperties(prefix = "breuninger.togglz")
@Validated
public class TogglzProperties {

  private @Min(0) int cacheTtl = 5000;
  private @Valid Console console = new Console();

  @Getter
  @Setter
  public static class Console {

    private boolean enabled = true;
  }
}

File: togglz/src/main/kotlin/com/breuninger/boot/togglz/autoconfigure/TogglzProperties.kt
package com.breuninger.boot.togglz.autoconfigure

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated
import java.util.concurrent.TimeUnit

@ConfigurationProperties(prefix = "breuni.togglz")
@Validated
class TogglzProperties {

  var enabled = false

  val cache = Cache()

  class Cache {

    var timeToLive: Long = 30
    var timeUnit = TimeUnit.SECONDS
  }

  val mongo = Mongo()

  class Mongo {

    var enabled = false
  }
}
